docs/proposals/scheduling/20220701-schedule-gang.md

```diff
@@ -0,0 +1,341 @@
+the real scenario, which is different from community.
+
+## Motivation
+In AI scenarios, lots of jobs need gang scheduling. The community have lots of related implements such as `coescheduling` or `vocalno`.
+However, in practice we find that the above solutions do not meet the needs of the business in some places.
+
+### Compared with competitors
+
+#### Coescheduling
+1. `coescheduling` implement a new queue-sort interface and other methods to let one gang's pods get out of the queue in order as much as possible.
+If a pod failed to be scheduled, the requests that have been successfully scheduled in this round of gang scheduling cycle will be rolled back,
+and the remaining pods waiting for scheduling will be rejected in PreFilter check until this scheduling cycle passed. 
+For example, there is a gang requires 10 tasks to be scheduled, if first 5 tasks allocated, the 6th task failed to be scheduled,
+`coescheduling` will roll-back first 5 tasks and ignore the remaining 4 tasks in this gang scheduling cycle. `coescheduling` simply use a 
+global time interval to control the gang scheduling cycle. The first defect is that the uniform time interval will cause 
+some problems. If the time configuration is too long, it will lead to useless waiting; If the time configuration is too short, 
+it will lead to useless scheduling. Secondly, it is very difficult for a large job to meet all resource requests at one time. 
+This mechanism will lead to a very low probability of full resources, and eventually make the job starve to death. We call this process as `strict-mode`.
+
+2. Some jobs have complex gang requirements. For example, a job has several roles. Each role will have several pods 
+and its own gang conditions. Jobs also need different roles to form different gang groups. All pods in a gang group can 
+trigger the bind process only after all roles in a gang group meet their gang conditions. The `coescheduling` can't meet
+this requirement.
+
+#### Volcano
+Volcano uses JobInfo as the basic unit for scheduling and is naturally friendly to gang. However, the scheduler is very 
+different from the community from the data structure to the process, which lead negative compatibility with the native k8s community.
+
+### Goals
+1. Define API to announce gang-scheduling-configuration.
+
+2. Provides a scheduler plugin to achieve gang-scheduling ability.
+
+### Non Goals and Future Work
+1. Provide ability to solve gang-resource-deadlock problems with `non-strict-mode`.
+
+## Proposal
+### API
+We recommend users declaring gang-scheduling-configuration by pod's annotation. First Reason, high level operator has no need 
+to maintain gang-crd's life circle like handling `update/create/delete` events. Second Reason, from a Scheduler perspective, 
+it's inconvenient to maintain receive-order-issue's between gang-crd and pod. According to practical experience, pod's annotation 
```

Comment:
Can you give more details about `receive-order-issue`? Why use the annotation form to declare Gang without this problem?